dz9/z1.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+import csv
+import datetime
+import json
+import math
+import os.path
+from random import randint as rnd
+from typing import Callable
+
+
+def deco_csv(function: Callable):
+    create_csvfile()
+
+    def wrapper():
+        with open('f.csv', 'r', encoding='UTF-8') as file:
+            data = csv.reader(file, quoting=csv.QUOTE_NONNUMERIC)
+            for coef in data:
+                if coef and coef[0] != 0:
+                    function(*coef)
+
+    return wrapper
+
+
+def json_result(func: Callable):
+    result = {}
+    if os.path.exists('f1.json'):
+        with open('f1.json', 'r', encoding='UTF-8') as file:
+            result = json.load(file)
+    else:
+        with open('f1.json', 'w', encoding='UTF-8') as file:
+            json.dump(result, file) \
```

Comment:
Наверное, правильнее будет та
```suggestion
    if os.path.exists('f1.json'):
        with open('f1.json', 'r', encoding='UTF-8') as file:
            result = json.load(file)
    else:
        result = {}
```